index.js

```diff
@@ -17,14 +13,33 @@ async function main() {
     });
 
     const token = core.getInput('token');
+    const source = core.getInput('source');
+    const origin = core.getInput('origin');
+
+    var args = [
+        'run', '--show-diff-on-failure', '--color=always'
+    ];
+
+    if (source) {
+        args.push('--source', source);
+    }
+    
+    if (origin) {
+        args.push('--origin', origin);
+    }
```

Comment:
since they're both required maybe:

```js
if (!!source != !!origin) {
    // error about both being required
} elif (...) {
   // source + origin
} else {
    // all-files
}
```

Reply:
I would prefer to keep them separate since one day, `pre-commit` might support open range with `--origin` only 🤞

Reply:
what would that even mean?

Reply:
If you don't specify origin it would mean `HEAD`